src/data/finsupp/basic.lean

```diff
@@ -1150,6 +1155,21 @@ finset.subset.trans support_sum $
   finset.subset.trans (finset.bUnion_mono $ assume a ha, support_single_subset) $
   by rw [finset.bUnion_singleton]; exact subset.refl _
 
+@[simp]
+lemma map_domain_sum_index {X Y : Type*} (f : X → Y)
```

Comment:
This lemma is almost identical to `sum_map_domain_index`, and differs only in the fact that `(h : Y → M →+ N)` is bundled. Can you:
* reorder the arguments to match `sum_map_domain_index`
* rename to something like `sum_map_domain_index_add_hom`?

Reply:
Ah! I was surprised I couldn't find this lemma already. I've followed your suggestions.

Curiously, it isn't possible to write an analogous lemma for `monoid_hom`, because even in the multiplicative case `M` is an `add_comm_monoid` still.

Reply:
Yeah, finsupp has the additive identity ingrained in its definition (as the unsupported values), so a lot of statements about sums don't quite work on products.

Reply:
This lemma and the new induction lemma are no longer needed, right?

Reply:
This lemma is still needed, to establish naturality of the `hom_equiv` when we finally set up the adjunction.

I think you're right that the new induction lemma is not needed. I would be happy to leave it, as it feels useful to me, but happy to go either way.

Reply:
But isn't this lemma now identical to `sum_map_domain_index_add_monoid_hom` below?

Reply:
Oh!